### ISSUE TITLE
What is a JSON Schema and why is it useful? How do I create/use a JSON Schema?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/learn-json-schema#1]
Answer the question that is _obviously_ on _everyone's_ mind... 😉

**Comment by sohilpandya** [comment]
### JSON SCHEMA

**JSON schema is a powerful tool for validating the structure of JSON data.**

_But what does it actually mean in layman's term?_

What is JSON?
- Javascript Object Notation
- store information.
- Uses following data structures
  - Object, Array, Number, String, Boolean, Null  

``` js
{  
    “key1”: “value1”,
    “key2”: “value2”,
}
```

#### What is a JSON schema?
- used to define the structure of the JSON Object.
- useful tool for defining what type of data structure you want to return.

**E.G.**  
When a user submits a form, you want to validate all the information they may have entered.

Expected Object:

``` js
{
    “type”: “object”,
    “properties”: {
        “first name”: { “type”: “string” },
        “last name”: { “type”: “string” },
        “age” : { “type”:  “number” }
    }
}
```

Now if the submit a the following object:  :heavy_multiplication_x: It would fail the validation that we have put in place.

``` js
{
    “first name”:”Mickey”,
    “last name”: “Mouse”
}
```

 But, the following object:  :heavy_check_mark: It would pass the validation.

``` js
{
    “first name”:”Mickey”,
    “last name”: “Mouse”,
    “age”: 100
}
```

**There we have our first JSON Schema for validation purposes! :smile:**

**Comment by deadcoder0904** [comment]
@sohilpandya edit **"first name"** to **"last name"** in the **Expected Object** as u have written it twice :smile:

**Comment by sohilpandya** [comment]
Thanks! @deadcoder0904 :smile:
